service_test.go

```diff
@@ -43,15 +45,14 @@ func TestServer_Run_Shutdown(t *testing.T) {
 	tests := []struct {
 		name    string
 		cp      Component
-		port    string
 		wantErr bool
 	}{
-		{name: "success", cp: &testComponent{}, port: "50000", wantErr: false},
-		{name: "failed to run", cp: &testComponent{errorRunning: true}, port: "50001", wantErr: true},
+		{name: "success", cp: &testComponent{}, wantErr: false},
+		{name: "failed to run", cp: &testComponent{errorRunning: true}, wantErr: true},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			os.Setenv("PATRON_HTTP_DEFAULT_PORT", tt.port)
+			os.Setenv("PATRON_HTTP_DEFAULT_PORT", getRandomPort())
```

Comment:
Question: why do we want the `getRandomPort` function and don't set a hard-coded number?

Reply:
The service does not have a mechanism to be stopped.
The error case stops successfully due to an error but the success will stop only if it is killed (SIGTERM etc.)
The test ends but the server does not shut down, and as a result, subsequent tests fail occasionally.
I had 2 options:
- Implement a shutdown feature, only for testing usage
- Randomize the ports so the new test run get's a new port
this is a hack but introducing code only for the test is not something I like either.
Any suggestions?

Reply:
Theoretically, `getRandomPort`  can return the same port twice we can avoid by check if the port is available.

* The following code adds extra overhead. *

something like:
```go
func getRandomPort() string {
	rnd:= 50000+rand.Int63n(10000)
	port := strconv.FormatInt(rnd, 10)
	ln, err := net.Listen("tcp", ":" + port)
	for  err != nil {
		rnd:= 50000+rand.Int63n(10000)
		port := strconv.FormatInt(rnd, 10)
		ln, err = net.Listen("tcp", ":" + port)
	}
	ln.Close()
	return port
}
``